code-name-taran/src/com/spaghettisoft/component/ninemenmorris/BoardState.java

```diff
@@ -0,0 +1,344 @@
+
+        initialize();
+    }
+
+    private static void initialize() {
+        for (int i = 0; i < boardArrays.length; i++) {
+            for (int j = 0; j < boardArrays[i].length; j++) {
+                if (boardArrays[i][j] == null) {
+                    boardArrays[i][j] = new BoardTile("*");
+                }
+            }
+        }
+        turnNumber = 1;
+    }
+
+    private void placePool(String poolType, String place) {
+
+        BoardTile tileToPlace = getBoardTileByPlace(place);
+
+        if (poolType.equals(BoardTile.WHITE) && (!checkForExistance(place))) {
+            tileToPlace.addWhitePool();
+        } else if (poolType.equals(BoardTile.BLACK) && (!checkForExistance(place))) {
+            tileToPlace.addBlackPool();
+        } else {
+            tileToPlace.emptyTile();
+        }
+        placedPools++;
+
+        // drawGame();
+
+        checkForMills(place);
+
+    }
+
+    private void movePool(String placeFrom, String placeTo) throws IllegalMoveException {
+
+        BoardTile tileFromPlace = getBoardTileByPlace(placeFrom);
+        BoardTile tileToPlace = getBoardTileByPlace(placeTo);
+
+        if (checkForFlying(placeFrom, placeTo)) {
+
+            tileToPlace.setPoolType(tileFromPlace.getPoolType());
+            tileFromPlace.emptyTile();
+            // drawGame();
+        } else {
+            throw new IllegalMoveException();
+        }
+
+        checkForMills(placeTo);
+
+    }
+
+    private static boolean checkForFlying(String placeFrom, String placeTo) {
+
+        // System.out.println(mapPossMovements.get(placeFrom));
+        boolean notFlying = mapPossMovements.get(placeFrom).contains(placeTo);
+
+        // if (notFlying == true) {
+        // System.out.println("This is true");
+        // } else {
+        // System.out.println("You are cheating. Try again!");
+        // checkForFlying(placeFrom, placeTo); // TOZI RED VEREN LI E?
+        // }
+        return notFlying;
+    }
+
+    private void deletePool(String placeFrom) {
+        BoardTile tileFromPlace = getBoardTileByPlace(placeFrom);
+
+        String deletedPoolType = tileFromPlace.getPoolType();
+
+        tileFromPlace.emptyTile();
+
+        drawGame();
+
+        checkForEndOfGame(deletedPoolType);
+
+    }
+
+    private boolean checkForMills(String place) {
+
+        boolean foundMills = false;
+        int millsCounter = 0;
+
+        BoardTile tilePlace = getBoardTileByPlace(place);
+        String currentPoolType = tilePlace.getPoolType();
+
+        for (int i = 0; i < mills.length; i++) {
+            for (int j = 0; j < mills[i].length; j++) {
+                if (place.equals(mills[i][j]) && currentPoolType == (getBoardTileByPlace(mills[i][0])).getPoolType()
+                        && currentPoolType == (getBoardTileByPlace(mills[i][1])).getPoolType()
+                        && currentPoolType == (getBoardTileByPlace(mills[i][2])).getPoolType()) {
+
+                    foundMills = true;

```

Comment:
When mills are found you can directly break out of the cycle. It is also better to separate the logic of checkeing of there are mills in different method. This will shrink the size of the method : )